File: code/java/src/main/java/fr/zomzog/prez/javaVsKotlin/java/PonyHandler.java
package fr.zomzog.prez.javaVsKotlin.java;

import fr.zomzog.prez.javaVsKotlin.java.data.Pony;
import org.springframework.stereotype.Service;
import org.springframework.web.reactive.function.server.ServerRequest;
import org.springframework.web.reactive.function.server.ServerResponse;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@Service
public class PonyHandler {
    public Mono<ServerResponse> findAll(ServerRequest serverRequest) {
        throw new RuntimeException("TODO");
    }
    public Mono<ServerResponse> findOne(ServerRequest serverRequest) {
        throw new RuntimeException("TODO");
    }
    public Mono<ServerResponse> create(ServerRequest serverRequest) {
        throw new RuntimeException("TODO");
    }

    public void doStuff(Pony p) {
        if (p == null) {
            throw new RuntimeException("Pony must not be null");
        }
        // stuff
    }
}

File: code/kotlin/src/main/kotlin/fr/zomzog/prez/javaVsKotlin/PonyHandler.kt
package fr.zomzog.prez.javaVsKotlin

import fr.zomzog.prez.javaVsKotlin.kotlin.data.Pony
import org.springframework.web.reactive.function.server.ServerRequest
import org.springframework.web.reactive.function.server.ServerResponse
import reactor.core.publisher.Mono

class PonyHandler {
    fun findAll(request: ServerRequest): Mono<ServerResponse> {
        TODO()
    }

    fun findOne(request: ServerRequest): Mono<ServerResponse> {
        TODO()
    }
    fun create(request: ServerRequest): Mono<ServerResponse> {
        TODO()
    }

    fun doStuff(p: Pony?): String {
        if (p == null) {
            throw RuntimeException("Pony must not be null")
        }
        return p.name
        // doStuff
    }

    fun String.asPony(): Pony = Pony(this)
}